client/app/components/_pickupList/pickupList.spec.js

```diff
@@ -0,0 +1,305 @@
+        1,
+        8
+      ],
+      "max_collectors": 2,
+      "store": 9,
+      "isUserMember": true,
+      "isFull": true
+    }
+  ];
+  
+  let storeData = {
+    "id": 9,
+    "name": "REWE Neuried",
+    "description": "dasdsd",
+    "group": 4,
+    "address": "Muenchen",
+    "latitude": 51.6180165487737,
+    "longitude": 2.8125
+  };
+
+  
+  describe("Module", () => {
+    // top-level specs: i.e., routes, injection, naming
+  });
+
+  describe("Controller", () => {
+    // controller specs
+    let controller;
+    
+    beforeEach(() => {
+      controller = $componentController("pickupList", {
+        $scope: $rootScope.$new()
+      }, {
+        storeId: 9,
+        header: "My amazing Pickups"
+      });
+    });
+    
+    
+    it("bindings", () => {
+      expect(controller.storeId).to.equal(9);
+      expect(controller.header).to.equal("My amazing Pickups");
+    });
+    
+    it("automatic update", () => {
+      $httpBackend.expectGET("/api/auth/status/").respond(authData);
+      $httpBackend.expectGET("/api/pickup-dates/?store=9").respond(pickupData);
+      $httpBackend.flush();
+    });
+    
+    
+    it("addPickupInfo functionality", () => {
+      $httpBackend.when("/api/auth/status/").respond(authData);
+      $httpBackend.when("/api/pickup-dates/?store=9").respond(pickupData);
+      $httpBackend.when("/api/stores/9/").respond(pickupData);
+      
+      controller.userId = 1;
+      controller.addPickuplistInfos(pickupData);
+      let updatedData = controller.allPickups;
+      expect(updatedData).to.deep.equal(pickupDataInfoAdded);
+      expect(updatedData[0].store).to.deep.equal(9);
+    });
+    
+    it("filter functionality", () => {
+      controller.allPickups = pickupDataInfoAdded;
+      controller.pickupList = {
+        showJoined: false,
+        showOpen: false,
+        showFull: true
+      };
+      expect(controller.filterPickups()).to.deep.equal(fullPickups);
+    });
+    
+    it("groupByDate functionality", () => {
+      expect(controller.groupByDate(pickupDataInfoAdded)).to.deep.equal(pickupDataInfoAddedGrouped);
+    });
+  });
+  
+  describe("Controller with showDetail = store", () => {
+    // controller specs
+    let controller;
+    
+    beforeEach(() => {
+      controller = $componentController("pickupList", {
+        $scope: $rootScope.$new()
+      }, {
+        storeId: 9,
+        header: "My amazing Pickups",
+        showDetail: "store"
+      });
+    });
+    
+    
+    it("addPickupInfo get Store Info functionality", () => {
+      $httpBackend.when("/api/auth/status/").respond(authData);
+      $httpBackend.when("/api/pickup-dates/?store=9").respond(pickupData);
+      $httpBackend.when("/api/stores/9/").respond(storeData);
+      
+      controller.userId = 1;
+      controller.addPickuplistInfos(pickupData);
+      let updatedData = controller.allPickups;
+      expect(updatedData[0].store).to.eventually.deep.equal(storeData);
+    });
+  });
```

Comment:
I got this working now, i have to figure out, whats the best way to contribute it here. Should i simply push?